extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -229,6 +230,7 @@ public function getAllAdminsData() {
 	 */
 	public function getRecentlyJoinedData() {
 		$recentlyJoined = $this->usersModel->getRecentlyJoinedUsers();
+		$recentlyJoined = $this->applyBadges( $recentlyJoined );
```

Comment:
so whole logic responsible for getting badge needs to be moved to CommunityPageSpecialUserModel